### PR TITLE
Oauth applications

### PR DESCRIPTION
**Pull request by 4p00rv**

### CODE REVIEW

File: src/javascript/binary/static/applications.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+pjax_config_page("user/applicationsws", function(){
+    return {
+        onLoad: function() {
+            if (!getCookieItem('login')) {
```
Comment (negar-binary) [inline src/javascript/binary/static/applications.js]: I think we have a new function for this, in page.js redirect_if_logout()

File: src/javascript/binary/static/applications/applications.ui.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    
+    function createTable(app){
+        Table.clearTableBody(tableID);
+        Table.appendTableBody(tableID, app, createRow);
+    }
+    
+    function createRow(data){
+        var name = data.name,
+            permissions = data.scopes.join(", "),
+            last_used = data.last_used ? moment.utc(data.last_used).format("Do MMMM YY") : text.localize("Never"),
```
Comment (negar-binary) [inline src/javascript/binary/static/applications/applications.ui.js]: I think you meant DD, not Do?

Comment (4p00rv) [inline src/javascript/binary/static/applications/applications.ui.js]: @negar-binary No, it is intended. This produce date like `March 22nd 16`.

Comment (negar-binary) [inline src/javascript/binary/static/applications/applications.ui.js]: I see, cool, thanks :)

**Comment by qingweibinary** [comment]
Can I know why you name it Applications?
Sounds too generic and make me wonder if Applications actually means multiple applications

File: src/javascript/binary/static/applications/applications.data.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+var ApplicationsData = (function(){
+    "use strict";
+
+    function getApplications(id){
+        var request = {oauth_apps: 1};
+        if(id){
+            $.extend(request,{revoke_app: id});
```
Comment (qingweibinary) [inline src/javascript/binary/static/applications/applications.data.js]: Why a get call is revoking permission?

File: src/javascript/binary/static/applications/applications.init.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                              )
+                          )
+                      );
+            }
+            ApplicationsUI.createTable(applications);
+        }
+    }
+    
+    function getNextBatch(){
+        ApplicationsData.getApplications();
```
Comment (qingweibinary) [inline src/javascript/binary/static/applications/applications.init.js]: this is a pretty useless functions, maybe you do not need it ?

File: src/javascript/binary/static/applications/applications.init.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        showLoadingImage($('<div/>', {id: 'loading'}).insertAfter('#applications-title'));
+        getNextBatch();
+    }
+    
+    function initTable(){
+        $("#applications-ws-container .error-msg").text("");
+        ApplicationsUI.clearTableContent();
+    }
+
+    function cleanPageState(){
```
Comment (qingweibinary) [inline src/javascript/binary/static/applications/applications.init.js]: This function is also seems redundant

File: src/templates/toolkit/account/applications.html.tt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<div class="grd-grid-12">
+    <div id="applications-ws-container">
+        <div class="page-title" id="applications-title">
+            <h1>Applications</h1>
```
Comment (qingweibinary) [inline src/templates/toolkit/account/applications.html.tt]: You might want to wrap it with `l` unless you will do translation from javascript

**Comment by 4p00rv** [comment]
@qingweibinary It actually displays all the oauth applications which were granted access, so named it applications.

File: src/templates/toolkit/account/applications.html.tt (hunk 6 above)
Comment (4p00rv) [inline src/templates/toolkit/account/applications.html.tt]: Yes I have it in javascript.

File: src/javascript/binary/static/applications/applications.data.js (hunk 3 above)
Comment (4p00rv) [inline src/javascript/binary/static/applications/applications.data.js]: @qingweibinary Well the revoke_app call also returns the applications. So based on response I wrapped it inside getApplications/

**Comment by qingweibinary** [comment]
@4p00rv 
I see
I still think Applications is not a good name, especially in global namespace, how about we rename it into AuthorizedApplications, or ConnectedApplications?

File: src/javascript/binary/static/applications/applications.data.js (hunk 3 above)
Comment (qingweibinary) [inline src/javascript/binary/static/applications/applications.data.js]: I suggest we split it, a get call with side effect is pretty surprising ~
if the id is never gonna be used, then let's omit it

**Comment by 4p00rv** [comment]
@qingweibinary I actually was confused too, but after I saw that **Twitter** also uses the same name for authorized applications, so I settled on it.

File: src/javascript/binary/static/applications/applications.ui.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            action = '';
+        var $row = Table.createFlexTableRow([name,permissions,last_used,action], columns,"data");
+        var $viewButtonSpan = Button.createBinaryStyledButton();
+        var $viewButton = $viewButtonSpan.children(".button").first();
```
Comment (nuruddeensalihu) [inline src/javascript/binary/static/applications/applications.ui.js]: .button from  `$viewButtonSpan.children(".button").first()` is css class right ? Is is it referring to button tag ? If its a css class , i suggest you avoid using the name that exist as html tag. If its a button tag , i don't think you would require the . or would you  ?

Comment (4p00rv) [inline src/javascript/binary/static/applications/applications.ui.js]: @nuruddeensalihu `Button.createBinaryStyledButton();` returns a button having class `button`. I'am just using a pre-written code. Also, the code has multiple usage of it, so changing the class would need refactoring in other places as well.

Comment (nuruddeensalihu) [inline src/javascript/binary/static/applications/applications.ui.js]: oic. :+1:

**Pull request merged**